### ISSUE TITLE
Injecting NSE's in specific paths in vl3

### ISSUE THREAD

**Issue by yuraxdrumz** [networkservicemesh/sdk#1300]
## Question

I am currently experimenting with vl3 and it seems to answer my requirement of NSC talking to another NSC.
I also have a requirement that, let's say every time I want to talk to a specific NSC I must pass a component, let's call it A.
Now I want my flow with vl3 to go NSC1 -> NSE1 -> A -> NSE2 -> NSC2.

If something like this is supported, that would be great. If not, I have two options I can think of, and correct me if I'm wrong:

1. Try doing composition with vl3, if it is even possible, and inject A as a passthrough
2. Catch `PathSegements` somewhere along the Registry -> NSMGR -> NSC and inject A as a new path segment

Basically, I want to be able to inject machine A anywhere in the line.

## Examples

An example would be NSC1 -> NSE1 -> A -> NSE2 -> NSC2.
![vl3-inject](https://user-images.githubusercontent.com/17418978/169698679-5622a2b9-4d21-4947-9374-7479e72febc7.png)

## Context

* Kubernetes Version: EKS 1.22
* Cilium CNI
* NSM 1.3.1

**Comment by yuraxdrumz** [comment]
I wanted to simplify my question, but i basically need something like in https://github.com/networkservicemesh/deployments-k8s/issues/5247
